### PR TITLE
IP-9519: Added best practice section to API docs

### PR DESCRIPTION
**Pull request by boedlen**

### CODE REVIEW

File: pages/backend-api--overview.md [hunk 1]
```diff
@@ -5,13 +5,39 @@ redirect_from: "/display/DOC/Backend+API"
...
 ## General Guidelines
 
 * If an API returns a humanly read message, do not rely on it. It may change over time without notification. The message is only intended to help you understand the return code.
 * Anything but a SUCCESS return code should be treated as an error. An ERROR return code may be changed with a more specific return code at a later stage, but a SUCCESS code will never be changed without previous notification.
 * Any breaking changes will either be notified in due time, or a new API version will be setup so the old one can run concurrently.
 * API methods may be appended without there being created a new overall API version, but only as long as they add non-breaking features like optional fields and extra return data. As a result of this, do use XPATH to query results, you should not rely on element index positions.
 
+## Best practice
+
+* **Keep your credentials save**
```
Comment (improvedk) [inline pages/backend-api--overview.md:25]: Should be "safe"

File: pages/backend-api--overview.md [hunk 2]
```diff
@@ -5,13 +5,39 @@ redirect_from: "/display/DOC/Backend+API"
...
 * If an API returns a humanly read message, do not rely on it. It may change over time without notification. The message is only intended to help you understand the return code.
 * Anything but a SUCCESS return code should be treated as an error. An ERROR return code may be changed with a more specific return code at a later stage, but a SUCCESS code will never be changed without previous notification.
 * Any breaking changes will either be notified in due time, or a new API version will be setup so the old one can run concurrently.
 * API methods may be appended without there being created a new overall API version, but only as long as they add non-breaking features like optional fields and extra return data. As a result of this, do use XPATH to query results, you should not rely on element index positions.
 
+## Best practice
+
+* **Keep your credentials save**  
+While our endpoint allows for you to use GET requests, we strongly suggest you use POST.
+Using GET requests will send any credentials as clear text over the internet ever if sent via HTTPS.
```
Comment (improvedk) [inline pages/backend-api--overview.md:27]: "even"

File: pages/backend-api--overview.md [hunk 3]
```diff
@@ -5,13 +5,39 @@ redirect_from: "/display/DOC/Backend+API"
...
+
+* **Rate limit your requests**  
+We recommend that you keep requests to our API below **150 per minute**. Some requests like for instance `Paper.GetAllPapers` will return the same result every time unless someone/something has made changes since the last request, so consider adding a short lived cache on your end, or only make calls when you know there there are changes.
```
Comment (improvedk) [inline pages/backend-api--overview.md:30]: Is "recommend" strongly worded enough, especially if we set the limit that high? Don't we need to send a message so our customers can clearly ascertain when using our API in unintended ways? I can't see many use cases where you'd need to many requests/sec, except if you don't cache locally. If the limit was say 50 requests per minute, then recommend might be OK as customers would clearly know they're going against recommendations by using up 150 requests/min.

File: pages/backend-api--overview.md [hunk 4]
```diff
@@ -5,13 +5,39 @@ redirect_from: "/display/DOC/Backend+API"
...
+
+* **Use caching where possible**  
+Depending on the endpoint you call in the API, we recommend that you cache responses when possible.
```
Comment (improvedk) [inline pages/backend-api--overview.md:33]: recommend => should

File: pages/backend-api--overview.md [hunk 5]
```diff
@@ -5,13 +5,39 @@ redirect_from: "/display/DOC/Backend+API"
...
+* **Rate limit your requests**  
+We recommend that you keep requests to our API below **150 per minute**. Some requests like for instance `Paper.GetAllPapers` will return the same result every time unless someone/something has made changes since the last request, so consider adding a short lived cache on your end, or only make calls when you know there there are changes.
+
+* **Use caching where possible**  
+Depending on the endpoint you call in the API, we recommend that you cache responses when possible.  
+Caching results that aren't prone to changes will make your software faster since you save a roundtrip to our API for data you already have.
+
+* **Use the proper abstractions**
+If you have a website or app that uses data from iPaper, we recommend that you create an endpoint in your backend that returns the data from iPaper.
+Having your own endpoint has allows you to manage caching, hide credentials, transform API results and much more.
```
Comment (improvedk) [inline pages/backend-api--overview.md:38]: Remove "has".

File: pages/backend-api--overview.md [hunk 6]
```diff
@@ -5,13 +5,39 @@ redirect_from: "/display/DOC/Backend+API"
...
+Using GET requests will send any credentials as clear text over the internet ever if sent via HTTPS.
+
+* **Rate limit your requests**  
+We recommend that you keep requests to our API below **150 per minute**. Some requests like for instance `Paper.GetAllPapers` will return the same result every time unless someone/something has made changes since the last request, so consider adding a short lived cache on your end, or only make calls when you know there there are changes.
+
+* **Use caching where possible**  
+Depending on the endpoint you call in the API, we recommend that you cache responses when possible.  
+Caching results that aren't prone to changes will make your software faster since you save a roundtrip to our API for data you already have.
+
+* **Use the proper abstractions**
```
Comment (improvedk) [inline pages/backend-api--overview.md:36]: I think we need to write explicitly that you should not call our API directly from any code that runs on user devices, e.g. apps, websites, etc. as this introduces the risk of end-users grabbing the credentials and misusing them.

**Review by cyberzed — APPROVED** [review]

**Pull request merged**